### DISCUSSION TITLE
Is there a different state for isValid form and full form?

### DISCUSSION THREAD

**Discussion by paufar** [react-hook-form/react-hook-form#4146] [Q&A]
Is there a way to know if all registered fields in a form are filled? 

A form is only valid if all required fields are validated, and fields are only validated if they are required.
If fields are required, the form does not submit. I want my form to have 2 states: 
1. all required fields are valid and 
2. all fields in the form (including custom fields) are filled and valid. 

[This code sandbox ](https://codesandbox.io/s/controller-rules-forked-okk7e?file=/src/App.tsx) shows a form with 1 required field and 2 non required field with validation. I want the user to always be able to submit the form but when the required field is valid, mark the form as valid and when all the fields are filled, mark the form as complete. 

Ideally the solution would work for large forms as the form i am working with is much larger than the example in the CSB so the solution would need to work with any number of registered fields.

@bluebill1049 Would appreciate your input on this 🙏

**Comment by wdfinch** [comment]
> I want the user to always be able to submit the form but when the required field is valid, mark the form as valid and when all the fields are filled, mark the form as complete.

1. Rather than using `handleSubmit` which checks if the form is valid, you could just have a button that calls a custom submit method and then use `getValues` to get all your form values. This would bypass validation.

2. Validation is designed to determine what fields are valid and therefore not allow submission if fields are invalid. I don't really understand why you would want to allow the user to submit validation fails and the library assumes this as well. If you do want to know which fields are filled or not without using validation, you will need to keep track of this yourself but manually checking the form values.
